### PR TITLE
test(checkbox): update demo html to match Figma

### PR DESCRIPTION
**Pull request by ManahariDahal**
**Related Issue:** #2508

## Summary
- updated calcite-checkbox encompassing the Figma design
- this version includes small, medium, and large _wrapped calcite label_ variants

### CODE REVIEW

**Review by julio8a — DISMISSED** [review]
Focus states: I don't think we need the `focused` variations as they are not actually focused. i.e. unchecked focused is the same as unchecked. The way to check focus is just to focus on the unchecked version. Same for others like checked focused, indeterminate unchecked focused, etc.

Aside from ^, this looks good to me!

**Comment by eriklharper** [comment]
> Focus states: I don't think we need the `focused` variations as they are not actually focused. i.e. unchecked focused is the same as unchecked. The way to check focus is just to focus on the unchecked version. Same for others like checked focused, indeterminate unchecked focused, etc.
> 
> Aside from ^, this looks good to me!

Why wouldn't we want to capture what focusing looks like?  IMO this would be a useful thing for Screener to capture once we get it running against demo pages.

**Comment by julio8a** [comment]
Maybe I missed something? @ManahariDahal, is there a difference between the focused ones? on load, visually, they look the same as the ones without the focused label.

**Comment by eriklharper** [comment]
> Maybe I missed something? @ManahariDahal, is there a difference between the focused ones? on load, visually, they look the same as the ones without the focused label.

My bad.  I'm forgetting that we converted the `focused` prop/attribute to a private state variable so applying `focused` doesn't do anything anymore.  Those examples should probably just get removed from the demo HTML and down the road when we get Screener setup to test against demo pages, we'll use the Screener Steps API to programmatically focus all the different variants and take screenshots of each in order to capture them properly.

The focused state for most if not all of the calcite designs is a blue focus ring 
image

**Comment by julio8a** [comment]
That sounds great!

**Comment by ManahariDahal** [comment]
> Focus states: I don't think we need the `focused` variations as they are not actually focused. i.e. unchecked focused is the same as unchecked. The way to check focus is just to focus on the unchecked version. Same for others like checked focused, indeterminate unchecked focused, etc.
> 
> Aside from ^, this looks good to me!

removed the focused states

**Review by julio8a — APPROVED** [review]
LGTM!

File: src/demos/calcite-checkbox.html [hunk 1]
```diff
@@ -32,846 +33,750 @@
...
+      **************************************************
+    -->
+    <div class="parent">
+      <div class="child right-aligned-text">wrapped in native label with for</div>
+
+      <!-- small -->
+      <div class="child">
+        <label for="light-wrapped-in-label-with-for">
+          <calcite-checkbox
+            id="light-wrapped-in-label-with-for"
```
Comment (caripizza) [inline src/demos/calcite-checkbox.html:716]: Can you make the checkbox ids and (wrapping label's `for` attributes) unique to each checkbox?

File: src/demos/calcite-checkbox.html [hunk 2]
```diff
@@ -32,846 +33,750 @@
...
+      **************************************************
+    -->
+    <div class="parent">
+      <div class="child right-aligned-text">sibling calcite-label</div>
+
+      <!-- small -->
+      <div class="child">
+        <calcite-label for="light-sibling-calcite-label"
+          >sibling calcite-label
+          <calcite-checkbox id="light-sibling-calcite-label" name="sibling-calcite-label" scale="s"></calcite-checkbox>
```
Comment (caripizza) [inline src/demos/calcite-checkbox.html:650]: Can you  make the ids unique here as well?

**Comment by bstifle** [comment]
1.  let's remove the `hovered` versions from the demo page, as we can just see those when interacting with them
2. ship it!

**Comment by julio8a** [comment]
I think leaving the hover will be good for Screener to check it as part of the automated testing.

**Comment by eriklharper** [comment]
> I think leaving the hover will be good for Screener to check it as part of the automated testing.

Hover testing is also something that Screeners Step API supports doing in tests which will be a more robust way of testing hover states since it will be testing against an actual browser hover event instead of just applying a custom `hovered` attribute.  

image

https://github.com/screener-io/screener-runner#testing-interactions

**Comment by julio8a** [comment]
Okay, let's remove the hover states and leverage screener

**Review by bstifle — APPROVED** [review]
nice work man

**Comment by ManahariDahal** [comment]
sweet! should I go ahead and merge then? @julio8a

**Comment by julio8a** [comment]
Go for it!

**Pull request merged**